seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -0,0 +1,177 @@
+        self.assertEqual(project.guid, response_json['projectGuid'])
+        self.assertListEqual(
+            [project.genome_version, project.description, project.workspace_namespace, project.workspace_name],
+            ['38', 'A test project', TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME])
+        mock_add_service_account.assert_called_with(user, TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME)
+        mock_email.assert_called_with(
+            subject='AnVIL data loading request',
+            body="""
+    Data from AnVIL workspace "{namespace}/{name}" needs to be loaded to seqr project <a href="http://testserver/project/{guid}/project_page">{name}</a> (guid: {guid})
+
+    The sample IDs to load are attached.    
+    """.format(namespace=TEST_WORKSPACE_NAMESPACE, name=TEST_NO_PROJECT_WORKSPACE_NAME, guid=project.guid),
+            to=['test_superuser@test.com', 'test_data_manager@test.com'],
+            attachments=[('{}_sample_ids.tsv'.format(project.guid), 'NA19675\nNA19678\nHG00735')]
+        )
+
+        # Test project exist
+        url = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME])
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Project "{name}" for workspace "{namespace}/{name}" exists.'
+                         .format(namespace=TEST_WORKSPACE_NAMESPACE, name=TEST_NO_PROJECT_WORKSPACE_NAME))
+
+        # Test sending email exception
+        url = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME2])
+        mock_email.side_effect = Exception('Something wrong while sending email.')
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        self.assertEqual(response.status_code, 200)
+        mock_api_logger.error.assert_called_with('Exception while sending email to user test_user_manager. Something wrong while sending email.')
+
+        # Test login locally
+        remove_token(user)  # The user will be same as logging in locally after the access token is removed
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/login/google-oauth2?next=/api/create_project_from_workspace/submit/my-seqr-billing/anvil-no-project-workspace2')
+
+
+class NoGoogleAnvilWorkspaceAPITest(AuthenticationTestCase):
+    fixtures = ['users']
+
+    def test_anvil_workspace_page(self):
+        url = reverse(anvil_workspace_page, args=[TEST_WORKSPACE_NAMESPACE, TEST_WORKSPACE_NAME])
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/login/google-oauth2?next=/workspace/my-seqr-billing/anvil-1kg%2520project%2520n%25C3%25A5me%2520with%2520uni%25C3%25A7%25C3%25B8de')
+
```

Comment:
you need to log a user in locally here (`self.login_base_user()`) and then test that it still redirects them. You shouldn't be testing the behavior of local log in in your anvil test case, you should test it here

Reply:
Good point.